_posts/pl/2021-03-xx-wspoldzielona-biblioteka-w-jenkins-pipeline.md

```diff
@@ -0,0 +1,124 @@
+---
+layout:    post
+title:     "Współdzielona biblioteka w jenkins pipeline"
+published: true
+lang:      pl
+date:      2021-03-xx 08:00:00 +0100
+author:    dkubiak
+image:     /assets/img/posts/2021-03-xx-xx/xx.jpg
+tags:
+- jenkins
+- devops
+- pipeline
+- shared library
+
+---
+W tym artykule zapoznamy się z mechanizmem bibliotek współdzielonych (Pipeline: Shared Groovy Libraries), które stanowią
+zależność do szeroko stosowanej wtyczki jenkinsowej pipelines (https://www.jenkins.io/doc/book/pipeline/).
+
+Każdy z nas podczas modelowania procesów CI/CD spotkał się z pewnymi podobieństwami pomiędzy projektami. Niewątpliwie
+jednym z takich procesów może być release projektu. W naszej firmie prawie każdy projekt budowany jest przy użyciu
+mavena. Tym samym release takich projektów jest procesem bardzo zunifikowanym. W związku z tym jest to idealnym
+kandydat, by zamknąć go w bibliotekę, ustawić na półkę i używać jak zajdzie taka potrzeba.
+
+Zakładając, że każdy projekt w procesie wydawniczym (pipeline) ma krok pod tytułem release. Wystarczy, że zbudujemy
+współdzieloną bibliotekę, którą użyjemy w kroku bez wchodzenia w sposób działania.
+
+Biblioteka ma zapewnić nam:
+
+- pobicie wersje w POMie do stabilnej
+- upload artefaktów (binarki) do repozytorium
+- ustalić nową wersję developerską + commit do głównej gałęzi.
+
+# Tworzenie biblioteki
+
+W nowo utworzonym repozytorium kodu tworzymy strukturę katalogów. W naszym przykładzie git@git.consdata/consdata-shared-lib
+
+```
```

Comment:
IMO trochę słabo to wygląda
![image](https://user-images.githubusercontent.com/1863167/112452680-29ecfb80-8d57-11eb-8092-36ea8f6f4e86.png)

Może obrazek?

Reply:
Zmniejszyłem rozmiar